include/onnxruntime/core/framework/op_kernel_context.h

```diff
@@ -176,10 +176,6 @@ class OpKernelContext {
   const OrtValue* GetImplicitInputMLValue(int index) const;
   OrtValue* GetOutputMLValue(int index);
 
-#ifdef ENABLE_TRAINING
-  Status SetOutputMLValue(int index, const OrtValue& ort_value);
-#endif
```

Comment:
I think we might still want to keep this method for now... I saw it used in @iK1D's PR #7328.

Reply:
Thanks for pointing this out. I have a meeting setup with him and few other folks that are using this API to understand their scenario and see if it can be solved via other means. My concern with this API is that is brings in outside tensors without doing any device checks that happens with graph inputs.

---
In reply to: [613610139](https://github.com/microsoft/onnxruntime/pull/7324#discussion_r613610139) [](ancestors = 613610139)

Reply:
IIUC, we still need this for the PythonOp introduced by the autograd.Function work. 

---
In reply to: [614569764](https://github.com/microsoft/onnxruntime/pull/7324#discussion_r614569764) [](ancestors = 614569764,613610139)

Reply:
We had a meeting and decided we can keep this API provided it is enhanced with appropiate device checks.